src/test/java/com/iexec/standalone/result/ResultServiceTests.java

```diff
@@ -293,6 +297,71 @@ void testSaveResultInfo() {
     }
 
+    @Test
+    void testGetResultModelWithZip() throws IOException {
+        ResultInfo resultInfo = mock(ResultInfo.class);
+        when(resultInfo.getImage()).thenReturn("image");
+        when(resultInfo.getCmd()).thenReturn("cmd");
+        when(resultInfo.getDeterministHash()).thenReturn("hash");
+
+        doReturn(resultInfo).when(spyResultService).getResultInfos(CHAIN_TASK_ID);
+        try (MockedStatic<Files> mockedFiles = mockStatic(Files.class)) {
+            mockedFiles.when(() -> Files.readAllBytes(any(Path.class))).thenReturn(new byte[]{1, 2, 3});
+
+            ResultModel resultModel = spyResultService.getResultModelWithZip(CHAIN_TASK_ID);
+
+            assertThat(resultModel).isNotNull();
+            assertThat(resultModel.getChainTaskId()).isEqualTo(CHAIN_TASK_ID);
+            assertThat(resultModel.getImage()).isEqualTo("image");
+            assertThat(resultModel.getCmd()).isEqualTo("cmd");
+            assertThat(resultModel.getZip()).isEqualTo(new byte[]{1, 2, 3});
+            assertThat(resultModel.getDeterministHash()).isEqualTo("hash");
+        }
+    }
+
+    @Test
+    void testGetResultModelWithZip_FileNotFound() throws IOException {
+        ResultInfo resultInfo = mock(ResultInfo.class);
+        when(resultInfo.getImage()).thenReturn("image");
+        when(resultInfo.getCmd()).thenReturn("cmd");
+        when(resultInfo.getDeterministHash()).thenReturn("hash");
+
+        doReturn(resultInfo).when(spyResultService).getResultInfos(CHAIN_TASK_ID);
+        try (MockedStatic<Files> mockedFiles = mockStatic(Files.class)) {
+            mockedFiles.when(() -> Files.readAllBytes(any(Path.class))).thenThrow(new IOException("File not found"));
+
+            ResultModel resultModel = spyResultService.getResultModelWithZip(CHAIN_TASK_ID);
+
+            assertThat(resultModel).isNotNull();
+            assertThat(resultModel.getChainTaskId()).isEqualTo(CHAIN_TASK_ID);
+            assertThat(resultModel.getImage()).isEqualTo("image");
+            assertThat(resultModel.getCmd()).isEqualTo("cmd");
+            assertThat(resultModel.getZip()).isEqualTo(new byte[0]);
+            assertThat(resultModel.getDeterministHash()).isEqualTo("hash");
+        }
+    }
+
+    @Test
+    void testCleanUnusedResultFolders() {
+        List<String> recoveredTasks = Arrays.asList("task1", "task2");
+        doReturn(Arrays.asList("task1", "task3")).when(spyResultService).getAllChainTaskIdsInResultFolder();
+        doReturn(true).when(spyResultService).purgeTask("task3");
+        spyResultService.cleanUnusedResultFolders(recoveredTasks);
+        verify(spyResultService, times(1)).purgeTask("task3");
+    }
+
+    @Test
+    void testGetAllChainTaskIdsInResultFolder_Empty() {
+        File mockFile = mock(File.class);
+        when(workerConfigurationService.getWorkerBaseDir()).thenReturn("/mock/path");
+        when(mockFile.list(any())).thenReturn(null);
```

Comment:
By default, calling any method on a mocked instance returns the default non-initialized value (0 for numeric values, `false` for booleans and `null` for objects). This line does not do anything, then.
By the way, you should probably use a real file in a temporary folder here as well.